### ISSUE TITLE
DicomTranscoder creates a tmp file

### ISSUE THREAD

**Issue by hashitha** [fo-dicom/fo-dicom#1351]
It seems that the following code uses a temp file, is there any way we can bypass this and do the conversion in memory?
I encountered this when converting a MultiFrame Dicom File with lots of frames

```
       var sharpImage = dicomImage.RenderImage(i).AsSharpImage();
 
        var jpegEncoder = new JpegEncoder
        {            
            Quality = 75
        };

        var jpegStream = new MemoryStream();
        await sharpImage.SaveAsJpegAsync(jpegStream, jpegEncoder);
```

![image](https://user-images.githubusercontent.com/10569142/157362694-d748eefe-efcc-477d-9e56-e2782f2ca842.png)

**Comment by aerik** [comment]
DicomImage will transcode a compressed image to uncompressed as necessary, but the transcoders will currently use a TempFileBuffer according to their own internal logic.  See the discussion in #557

**Comment by gofal** [comment]
Agree, it's not SharpImage that creates the tempfile, but it is the DicomTranscoder. So you would also get this tempfile if you would use fo-dicom.Imaging.Desktop and create Bitmaps.

**Comment by gofal** [comment]
related to #557 ?

**Comment by gofal** [comment]
The issue is to change the API to the transcoder, so that the application can decide if tempfiles may be used or if memory has to be used.

**Comment by aerik** [comment]
> The issue is to change the API to the transcoder, so that the application can decide if tempfiles may be used or if memory has to be used.

YES!  Exactly.  Make the transcoder only transcode one frame at a time so they always comfortably fit in any reasonable amount of memory, then deal with the frames is FO-Dicom.  (This is exactly what I did in my branch).

**Comment by jaime-olivares** [comment]
@aerik if you already have a solution to this request, you may share it as a Pull Request.

**Comment by aerik** [comment]
@jaime-olivares Thanks, but I'm working on a branch of 3.0, so I don't have code to make an appropriate pull request.
